logdna/logdna.py

```diff
@@ -100,27 +105,27 @@ def flush(self):
                         'tags': self.tags if self.tags else None},
                     stream=True,
                     timeout=self.request_timeout)
+                res.raise_for_status()
+                # when no RequestException happened
+                self.exception_flag = False
                 self.buf = []
                 self.buf_byte_length = 0
                 if self.flusher:
                     self.flusher.cancel()
                     self.flusher = None
-                self.lock.release()
                 # Ensure messages that could've dropped are appended back onto buf
                 self.buf = self.buf + self.secondary
                 self.secondary = []
+                self.lock.release()
         except requests.exceptions.RequestException as e:
             if self.flusher:
                 self.flusher.cancel()
                 self.flusher = None
+
             self.lock.release()
-            if not self.exception_flag:
-                self.exception_flag = True
-                if self.verbose in ['true', 'error', 'err', 'e']:
-                    logger.error('Error in Request to LogDNA: %s', str(e))
-        else:
-            # when no RequestException happened
-            self.exception_flag = False
+            self.exception_flag = True
+            if self.verbose in ['true', 'error', 'err', 'e']:
+                print('Error happened while trying to send the logs')
```

Comment:
Kind of weird language here, I'd say `Error sending logs`. And why are we not including the exception in the error line anymore?